dev/statement_serializer.h

```diff
@@ -1298,11 +1301,11 @@ namespace sqlite_orm {
                         if(columnIndex > 0) {
                             ss << ", ";
                         }
-                        if(column.member_pointer) {
-                            ss << serialize(object.*column.member_pointer, context);
-                        } else {
-                            ss << serialize((object.*column.getter)(), context);
-                        }
+#if __cplusplus >= 201703L  // C++17 or later
+                        ss << serialize(std::invoke(column.member_pointer, object), context);
```

Comment:
this API is very nice. I have one idea about reducing amount of `#ifdef`s. We can create `invoke` function which will call `std::invoke` with C++17 and higher and `access_field_value` with C++14. Then we can write such statements using one line not five. What do you think?

Reply:
You are absolutely right about the ugly `#ifdef`s, I was expecting your feedback :)
In this sense it's best if we polyfill `std::invoke`? It's doable with either overloads or reusing `static_if`. The polyfill only needs to be good enough for internal purposes.

Reply:
I was thinking about an overload of `invoke` inside `sqlite_orm` namespace with one `#ifdef` inside but you can use polyfill also